### PR TITLE
Simplify user config merge

### LINKED ISSUE #10: Simplify user config merge
I think we could simplify the configure function, because merging each property one by one is quite tedious

### PR DESCRIPTION
**Pull request by DaveKeehl**
Closes #10

### CODE REVIEW

**Review by dabico — CHANGES_REQUESTED** [review]

File: src/lettercrap.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export type Config = {
   update_interval: number;
 };
 
-const config: Config = {
+let config: Config = {
```
[the remark this answers is not in the document]
Comment (DaveKeehl) [reply src/lettercrap.ts:10]: Yeah that would work (pretty sure also in TS). Regarding the validations, we could create a function and call inside the forEach.

[the remark this answers is not in the document]
Comment (dabico) [reply src/lettercrap.ts:10]: Yeah, that will be sufficient IMO. The only thing we should take care of is the validation for values like `font_family` and `font_weight`. I think we may have to introduce or re-use some special types that are appropriate for those style settings, as they can't be any string.

[the remark this answers is not in the document]
Comment (dabico) [reply src/lettercrap.ts:10]: For the other values:

- `content` can be any string value, so long as it's not `null` or `undefined`;
- `letters` should be a non-blank string of _printable_ characters;
- `words` can be any array of strings, so long as it's not `null` or `undefined`;
- `update_interval` should be a positive integer.

What do you think?

[the remark this answers is not in the document]
Comment (DaveKeehl) [reply src/lettercrap.ts:10]: > For the other values:
> 
> * `content` can be any string value, so long as it's not `null` or `undefined`;
> * `letters` should be a non-blank string of _printable_ characters;
> * `words` can be any array of strings, so long as it's not `null` or `undefined`;
> * `update_interval` should be a positive integer.
> 
> What do you think?

Sounds good. Some validations will have to be done at runtime though (e.g. check for positive integers, check for non-blank printable characters). We can either do it by hand or by bringing in a validation library like [Zod](https://zod.dev/), which is pretty popular.

[the remark this answers is not in the document]
Comment (DaveKeehl) [reply src/lettercrap.ts:10]: > We should base it on the MDN documentation:
> 
> * https://developer.mozilla.org/en-US/docs/Web/CSS/font-family
> * https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight
> 
> In the case of `font_family`, we should probably restrict ourselves to the generic font families, and in the case of `font_weight` the named values (`normal`, `bold`, `bolder` and `lighter`) will suffice.

For font weight I agree it should be enough. However, I'm a bit hesitant regarding font family. I think the user should be free to use any font.

**Review by dabico — APPROVED** [review]

**Review by dabico — APPROVED** [review]

**Pull request merged**